src/Cli/dotnet/commands/dotnet-tool/install/ParseResultExtension.cs

```diff
@@ -36,6 +30,13 @@ public static VersionRange GetVersionRange(this ParseResult parseResult)
             }
 
             VersionRange versionRange = null;
+
+            // accept 'bare' versions and interpret 'bare' versions as NuGet exact versions
+            if (!string.IsNullOrEmpty(packageVersion) && SemanticVersion.TryParse(packageVersion, out SemanticVersion version2))
+            {
+                packageVersion = "[" + packageVersion + "]";
```

Comment:
This works, but I think I'd prefer if instead of adding the brackets here, we change the logic in ToolPackageDownloader and NuGetPackageDownloader to detect whether a specific version was specified differently.  It would look something like this I believe:

```c#
if (versionRange.MinVersion != null && !versionRange.IsFloating &&
    (versionRange.MaxVersion == null || versionRange.MinVersion == versionRange.MaxVersion))
```

@nkolev92 @zivkan Does that logic look like it makes sense?

Reply:
You'll lose the ability to differentiate between someone who enters `--version 1.2.3` and `--version "[1.2.3, )"`, and so will "incorrectly" do an exact match for someone who very explicitly appears to be asking for "the lowest version greater than or equal to 1.2.3".

Also, if you don't check `IsMinInclusive`, then you'll incorrectly detect "version greater than, but not equal to X" `(1.2.3, )`: https://github.com/NuGet/NuGet.Client/blob/c5abfc6291526cf83365f86bdf03319ee3123180/test/NuGet.Core.Tests/NuGet.Versioning.Test/VersionRangeTests.cs#L480-L491. But the likelyhood of anyone wanting this is so low as to probably be not relevant.

I don't think that range `[1.2.3, ]` makes sense. If `VersionRange.Parse` doesn't throw, then technically this isn't the same thing as `1.2.3`, so technically `IsMaxInclusive` should be checked. But now this is getting unhelpfully pedantic, so it probably doesn't matter if this get treated like an exact match

Finally, if you keep this version parsing here, you've already done most of the expensive CPU work, so you could `return new VersionRange(minVersion: packageVersion, isMinInclusive: true, maxVersion: packageVersion, isMaxInclusive: true, originalString = "[" + packageVersion + "]");`

Reply:
Thank you very much for suggestions @dsplaisted and @zivkan